src/node_url.h

```diff
@@ -492,7 +492,7 @@ enum url_error_cb_args {
 #define XX(name) name,
   ERR_ARGS(XX)
 #undef XX
-} url_error_cb_args;
+};
```

Comment:
Yes, when `node_url.h` is included in the `cctest/test_url.h`, a compile error occurs because the `url_error_cb_args` is defined multiple times. I had to make similar changes to the other structs defined in this file. It would likely be good to refactor the `node_url.h` file a bit if we intend for it to be generally includable in more than just `node_url.cc`.